master/internal/core.go

```diff
@@ -374,56 +394,22 @@ func (m *Master) Run() error {
 
 	webuiRoot := filepath.Join(m.config.Root, "webui")
 	reactRoot := filepath.Join(webuiRoot, "react")
+	reactIndex := filepath.Join(reactRoot, "index.html")
 
 	// Docs.
 	m.echo.Static("/docs/rest-api", filepath.Join(webuiRoot, "docs", "rest-api"))
 	m.echo.Static("/docs", filepath.Join(webuiRoot, "docs"))
 
-	type fileRoute struct {
-		route string
-		path  string
-	}
-
-	// React WebUI.
-	reactIndexFiles := [...]fileRoute{
-		{"/", "index.html"},
-		{"/det", "index.html"},
-		{"/det/*", "index.html"},
-	}
-
-	reactFiles := [...]fileRoute{
-		{"/security.txt", "security.txt"},
-		{"/.well-known/security.txt", "security.txt"},
-		{"/color.less", "color.less"},
-		{"/manifest.json", "manifest.json"},
-		{"/favicon.ico", "favicon.ico"},
-		{"/favicon.ico", "favicon.ico"},
-	}
-
-	reactDirs := [...]fileRoute{
-		{"/favicons", "favicons"},
-		{"/fonts", "fonts"},
-		{"/static", "static"},
-		{"/wait", "wait"},
-	}
-
-	for _, indexRoute := range reactIndexFiles {
-		reactIndexPath := filepath.Join(reactRoot, indexRoute.path)
-		m.echo.GET(indexRoute.route, func(c echo.Context) error {
-			c.Response().Header().Set("Cache-Control", "no-cache, no-store, must-revalidate")
-			c.Response().Header().Set("Pragma", "no-cache")
-			c.Response().Header().Set("Expires", "0")
-			return c.File(reactIndexPath)
-		})
-	}
-
-	for _, fileRoute := range reactFiles {
-		m.echo.File(fileRoute.route, filepath.Join(reactRoot, fileRoute.path))
-	}
-
-	for _, dirRoute := range reactDirs {
-		m.echo.Static(dirRoute.route, filepath.Join(reactRoot, dirRoute.path))
-	}
+	webuiGroup := m.echo.Group(webuiBaseRoute)
+	webuiGroup.File("/", reactIndex)
+	webuiGroup.GET("/*", func(c echo.Context) error {
+		groupPath := strings.TrimPrefix(c.Request().URL.Path, webuiBaseRoute+"/")
+		requestedFile := filepath.Join(reactRoot, groupPath)
+		if fileExists(requestedFile) {
```

Comment:
I need to check whether this would open us up to directory traversal attack issues. we might need to sanitize the path

Reply:
I added some counter measures